Leanplum/Manager/LPRequestQueue.m

```diff
@@ -11,24 +11,40 @@
 #import "LPAPIConfig.h"
 #import "LPConstants.h"
 #import "LPRequestManager.h"
+#import "LPRequestCallbackManager.h"
+
+@interface LPRequestQueue()
+
+@property (nonatomic, strong) LPRequestCallbackManager *callbackManager;
+
+@end
 
 @implementation LPRequestQueue
 
 + (instancetype)sharedInstance {
     static LPRequestQueue *sharedManager = nil;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        sharedManager = [[self alloc] init];
+        sharedManager = [[self alloc] initWithCallbackManager:[LPRequestCallbackManager sharedManager]];
     });
     return sharedManager;
 }
 
+- (id)initWithCallbackManager:(LPRequestCallbackManager *)callbackManager {
+    self = [super init];
+    if (self) {
+        _callbackManager = callbackManager;
+    }
+    return self;
+}
+
 - (void)enqueue:(LPRequest *)request {
     @synchronized (self) {
         NSMutableDictionary *args = [self createArgsDictionaryForRequest:request];
         [LPRequestManager addRequest:args];
-        
-        //Add callback code here.
+        [self.callbackManager add:request.requestId
+                        onSuccess:request.successResponse
+                        onFailure:request.failureResponse];
     }
 }
 
```

Comment:
You also need to think on sendRequests. So, on success, you will need to go through each request call back. And make those call backs.